packages/backend/src/setup/createRpcTransactionUpdaters.ts

```diff
@@ -43,18 +45,28 @@ export function createRpcTransactionUpdaters(
   return rpcUpdaters
 }
 
-function createL2Provider(rpc: RpcTransactionApi, alchemyApiKey: string) {
-  switch (rpc.provider) {
-    case 'alchemy':
-      return new providers.AlchemyProvider(rpc.networkName, alchemyApiKey)
-
-    case 'jsonRpc':
-      return new providers.JsonRpcProvider({
-        url: rpc.url,
-        timeout: 10000,
-      })
+function createL2Provider(
+  rpc: RpcTransactionApi,
+  projectId: ProjectId,
+  config: Config,
+) {
+  if (rpc.provider === 'jsonRpc') {
+    return new providers.JsonRpcProvider({
+      url: rpc.url,
+      timeout: 10000,
+    })
+  }
 
-    default:
-      throw new Error('Unknown provider')
+  let apiKey = ''
+  if (projectId === ProjectId('arbitrum')) {
+    apiKey = config.optimismAlchemyApiKey
+  }
+  if (projectId === ProjectId('optimism')) {
+    apiKey = config.optimismAlchemyApiKey
   }
+  if (!apiKey) {
```

Comment:
Shouldn't this fall back to `config.alchemyApiKey`?

Reply:
Maybe we don't want to flood this key with ton of other requests. I think it would make sense to create a key for each project in that case